app/controllers/application_controller.rb

```diff
@@ -89,9 +90,41 @@ def set_hompepage_nav
   end
 
   def send_csv_cases(action_string)
+    specific_report = params[:report]
+    if specific_report
+      send_csv_case_by_specific_report(specific_report)
+    else
+      send_csv_case_by_default(action_string)
+    end
+  end
+
+  def get_avaliable_general_reports
+    if current_user
+      @general_available_reports =  Pundit.policy_scope(current_user, get_general_reports)
+    else
+      []
+    end
+  end 
+
+  def get_general_reports
+    ReportType.where("abbr like ? ","R90%").all
+  end
+
+  def send_csv_case_by_specific_report(specific_report)
+    headers["Content-Type"] = 'text/csv; charset=utf-8'
+    report_type = ReportType.find_by_abbr(specific_report)
+    report_service_class = report_type.class_name.constantize
+    report_service = report_service_class.new(case_scope: @cases)
+    headers["Content-Disposition"] =
+      %(attachment; filename="#{report_service.filename}")
+      self.response_body = report_service.to_csv()
+  end
+
+  def send_csv_case_by_default(action_string)
     headers["Content-Type"] = 'text/csv; charset=utf-8'
     headers["Content-Disposition"] =
       %(attachment; filename="#{CSVGenerator.filename(action_string)}")
-    self.response_body = CSVGenerator.new(@cases)
+      self.response_body = CSVGenerator.new(@cases, CSVExporter.new(nil))
   end
+
 end
```

Comment:
This seems not quite general enough for application controller. I think `@general_available_reports` is only used by `_case_list.html.slim` maybe?

Wondering if we could put these changes into a controller concern and include it from app/cases/searches_controller and app/cases/filters_controller (and any others) instead of adding this to the application controller?

Tiny thing too, but the two adjectives in "general_available_reports" seems a bit overly specific - there aren't any other reports in this scope to get confused with, I think. Maybe just call this instance var "available_reports"?